src/infra/team-maintenance.md

```diff
@@ -4,34 +4,64 @@ The roster of the Rust teams is always in flux. From time to time, new people
 are added, but also people sometimes opt to into "alumni status", meaning that
 they are not currently an active part of the decision-making process.
 Unfortunately, whenever a new person is added or someone goes into alumni
-status, there are a number of disparate places that need to be updated. This
-page aims to document that list. If you have any questions, or need someone with
-more privileges to make a change for you, a good place to ask is `#infra` on
-Discord.
+status, there are a number of disparate places that need to be updated.
 
-### Team repo
+# Team repo
 
 Membership of teams is primarily driven by the config files in the
-[rust-lang/team repo][team repo]. Several systems use the team repo data to
-control access:
+[rust-lang/team repo][team repo]. See the README of that respository for the
+systems integrated with it.
 
-- the [team website]
-- bors r+ rights
-- rfcbot interaction
-- Mailgun email lists
+# Rules for changes to team repo
 
-Team membership is duplicated in a few other places listed below, but the
-long-term goal is to centralize on the team repo.
+Pull requests to the repository are merged by the [team-admins], who use these rules to merge PRs:
+
+### `people`, `teams`, and `repos` directories
+
+If a change is related to an individual and does not expand permissions, then only the individual's approval is required. If the change has already been made outside of the team repo (e.g., GitHub username change) then it is considered implicitly approved. This non-exhaustively includes:
```

Comment:
> and does not expand permissions

I'm not sure it's always easy to tell when permissions are impacted or not. Like some marker teams are just markers for notifications but others have permissions associated with them. Are we running the risk of someone thinking a change doesn't impact permissions but in reality it does?

Reply:
Ideally, everything ends up being tied in to the team repo long-term so this becomes a CI check or similar... but I'm not sure we can get there quickly.

My sense is that the answer probably should be organizational for now - i.e., don't tie permissions to teams that aren't obviously permission-d. When in doubt checking with infra and/or escalating to a lead seems ok.

Do you feel like this comes up in practice? I feel like I'm usually pretty confident in what the implications of a change to team repo are. Maybe that's false confidence though :)

Reply:
I similarly feel pretty confident, but I echo your desire to get to just a CI check in the long term and this is just me worrying about the time between now and then.

Reply:
I'm going to go ahead and mark this thread resolved. I think we want the CI check anyway (no need for that to be coupled here) and it should be OK in practice in the meantime. Folks can always fallback on the more expensive "get team lead to approve" strategy.